File: task2_loops.py
# ISE102 Assessment 1
# Student: Aayush Parashar  |  ID: A00167472
# Task 2 – Loops

"""Program: multiplication_table.py
Description: Print a formatted multiplication table up to N.
"""

def print_table(n: int) -> None:
    for i in range(1, n + 1):
        row = []
        for j in range(1, n + 1):
            row.append(f"{i*j:4}")
        print("".join(row))

def main() -> None:
    try:
        n = int(input("Table size (1‑20): "))
        if not 1 <= n <= 20:
            raise ValueError
        print_table(n)
    except ValueError:
        print("Please enter an integer between 1 and 20.")

if __name__ == "__main__":
    main()

File: task2_loops.cpp
// ISE102 Assessment 1
// Student: Aayush Parashar  |  ID: A00167472
// Task 2 – Loops (C++ Version)

/*
Program: multiplication_table.cpp
Description: Print a formatted multiplication table up to N.
*/

#include <iostream>
#include <iomanip>

void print_table(int n) {
    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            std::cout << std::setw(4) << i * j;
        }
        std::cout << std::endl;
    }
}

int main() {
    try {
        int n;
        std::cout << "Table size (1‑20): ";
        std::cin >> n;
        
        if (std::cin.fail()) {
            throw std::runtime_error("Invalid input");
        }
        
        if (n < 1 || n > 20) {
            throw std::runtime_error("Out of range");
        }
        
        print_table(n);
        
    } catch (const std::exception& e) {
        std::cout << "Please enter an integer between 1 and 20." << std::endl;
    }
    
    return 0;
} 